### ISSUE TITLE
DBM Announce trigger type not triggering as of DBM 2.5.31

### ISSUE THREAD

**Issue by garrymullins** [WeakAuras/WeakAuras2#3575]
### Description

When using a trigger type of "Other Addons" with sub-type of "DBM Announce", the trigger no longer fires when a DBM Announce event occurs. 

This trigger type worked with DBM 2.5.30 and prior, but as of 2.5.31 it no longer triggers. 

The easiest way to test this is to zone into Hyjal Summit and start the Hyjal event, which fires a DBM Announce event but the DBM Announce trigger type no longer detects it. 

![image](https://user-images.githubusercontent.com/6465680/161437342-08dd809c-8e6a-45fb-94a2-44d63058a708.png)

### WeakAuras Version

3.7.16

### World of Warcraft Flavor

The Burning Crusade

### Tested with only WeakAuras

- [X] Yes
- [ ] No

### Lua Error

Normally with the DBM Announce trigger type defined in a trigger you can see the DBM Announce event firing in "/eventtrace", but you don't see this.

### Reproduction Steps

**Proper Behaviour**
1. Zone into Hyjal Summit (or some other instance that has easily accessible DBM Announce events for testing) with DBM 2.5.30 installed and the latest version of WeakAuras. 
2. Create a WA with an "Other Addons" > "DBM Announce" trigger type defined. For testing purposes select Message and Contains, and then just type in "Wave" for the message. You can set a duration to see that the WA triggers.
3. Activate the Hyjal event (or receive some DBM Announce message, wherever your chosen testing environment is)
4. The WA should trigger when the DBM Announce event fires
5. Additionally, when the DBM Announce event occurs, you should be able to see the "DBM_Announce" event under /eventtrace. 

![image](https://user-images.githubusercontent.com/6465680/161437676-529e823f-5a59-4bcd-8402-2032bb2c709f.png)

**Failure Scenario**
1. Update to DBM 2.5.31 and repeat above steps, you will see the WA does not trigger and the "DBM_Announce" event does not show under /eventtrace

### Last Good Version

3.7.16

**Comment by garrymullins** [comment]
I figured out the problem. Near line 7740 of DBM-Core.lua, the code is checking for a bunch of sound-related conditions and if those conditions are met, it returns the function and therefore doesn't proceed to fire the DBM_Announce event below. 

For my own purposes, temporarily as a work-around, I moved the fireEvent("DBM_Announce" ... ) line above this IF statement that is exiting the function.
